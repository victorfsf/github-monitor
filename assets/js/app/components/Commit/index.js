import Commit from './Commit';

export default Commit;
