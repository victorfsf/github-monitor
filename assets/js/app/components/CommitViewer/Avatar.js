import React from 'react';
import PropTypes from 'prop-types';
import octocat from 'app/images/github/octocat.png';
import { GITHUB_AVATAR_URL } from 'constants/github';


const Avatar = props => (
  <img
    src={props.id ? `${GITHUB_AVATAR_URL}${props.id}?v=4` : octocat}
    className="mr-3 rounded"
    alt={props.name}
    width="40px"
  />
);


Avatar.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string.isRequired,
};


Avatar.defaultProps = {
  id: null,
};


export default Avatar;
