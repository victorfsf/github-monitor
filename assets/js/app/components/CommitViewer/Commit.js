import React from 'react';
import PropTypes from 'prop-types';
import Box from 'app/components/Box';
import octocat from 'app/images/github/octocat.png';
import { commitPropTypes } from 'types';
import { formatDate } from 'utils';
import { GITHUB_URL } from 'constants/github';
import { Link } from 'react-router-dom';


const Commit = (props) => {
  const { payload } = props;
  const message = payload.message.split('\n', 1)[0];
  const avatar = (
    <img
      src={payload.avatar || octocat}
      className="mr-3 rounded"
      alt={payload.author}
      width="40px"
    />
  );
  return (
    <Box>
      <div className="media">
        {payload.login ? (
          <a href={`${GITHUB_URL}${payload.login}`} target="blank_">
            {avatar}
          </a>
        ) : avatar}
        <div className="media-body">
          <div className="row">
            <div className="col-8">
              <h6 title={message}>
                {message}
              </h6>
            </div>
            <div className="col-4">
              <small className="pull-right text-muted">
                {formatDate(payload.date)}
              </small>
            </div>
          </div>
          <small className="row">
            <div className="col-8">
              {payload.author}
              <i className="fa fa-code-fork px-2" />
              {payload.branch}
              <i className="fa fa-angle-right px-2" />
              <a href={payload.url}>{payload.sha}</a>
            </div>
            <div className="col-4">
              <span className="pull-right">
                <Link to={`/commits/${payload.repository}`}>
                  {payload.repository}
                </Link>
              </span>
            </div>
          </small>
        </div>
      </div>
    </Box>
  );
};


Commit.propTypes = {
  payload: PropTypes.shape(commitPropTypes),
};


Commit.defaultProps = {
  payload: null,
};


export default Commit;
