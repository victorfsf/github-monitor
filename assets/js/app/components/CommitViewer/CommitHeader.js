import React from 'react';
import PropTypes from 'prop-types';
import Box from 'app/components/Box';
import { GITHUB_URL } from 'constants/github';
import { formatDate } from 'utils';


const CommitHeader = props => (
  <Box>
    <div className="align-middle">
      <div className="row">
        <div className="col-sm-12 col-md-6">
          Captured Commits
          <button
            type="button"
            className="btn btn-sm btn-link"
            onClick={props.onClick}
          >
            <i className="fa fa-refresh" />
          </button>
          {props.repo && (
            <small>
              <a href={`${GITHUB_URL}${props.repo}`} target="blank_">
                {props.repo}
              </a>
            </small>
          )}
        </div>
        <div className="col-sm-12 col-md-6">
          <small className="float-md-right">
            {props.count} Commits
          </small>
        </div>
      </div>
      <div className="row">
        <div className="col-12">
          <small className="text-muted">
            Last Update: {
              props.lastUpdate ?
              formatDate(props.lastUpdate) :
              'Loading...'
            }
          </small>
        </div>
      </div>
    </div>
  </Box>
  );


CommitHeader.propTypes = {
  onClick: PropTypes.func.isRequired,
  lastUpdate: PropTypes.string,
  repo: PropTypes.string,
  count: PropTypes.number,
};


CommitHeader.defaultProps = {
  repo: '',
  lastUpdate: null,
  count: 0,
};


export default CommitHeader;
