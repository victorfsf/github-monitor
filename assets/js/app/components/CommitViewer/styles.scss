.media {
    .media-body {
        .title {
            font-size: 0.9em;
            font-weight: 600;
        }
    }

    .avatar {
        background-color: #f0f0f0;
        width: 40px;
        height: 40px;
        color: #cacaca;
        i {
            line-height: 40px;
        }
    }
}
