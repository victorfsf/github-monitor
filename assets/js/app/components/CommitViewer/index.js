import CommitViewer from './CommitViewer';

export default CommitViewer;
