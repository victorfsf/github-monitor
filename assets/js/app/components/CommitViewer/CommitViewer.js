import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { commitPropTypes, breadcrumbPropTypes } from 'types';
import { fetchCommitsIfNeeded } from 'redux/actions/commits';
import { Container, Spinner, Pagination } from 'app/components';
import Commit from './Commit';
import CommitHeader from './CommitHeader';


class CommitViewer extends React.Component {

  componentDidMount() {
    const { dispatch, repo } = this.props;
    dispatch(fetchCommitsIfNeeded(null, false, repo));
  }


  onClickPrev() {
    const { dispatch, prevPage, history, repo } = this.props;
    dispatch(fetchCommitsIfNeeded(prevPage, true, repo));
    history.push({ search: prevPage });
  }

  onClickNext() {
    const { dispatch, nextPage, history, repo } = this.props;
    dispatch(fetchCommitsIfNeeded(nextPage, true, repo));
    history.push({ search: nextPage });
  }

  handleRefresh() {
    const { dispatch, repo } = this.props;
    dispatch(fetchCommitsIfNeeded(null, true, repo));
  }

  render() {
    const {
      commits, isFetching, lastUpdate,
      prevPage, nextPage, breadcrumbs,
      count, repo,
    } = this.props;
    return (
      <Container breadcrumbs={breadcrumbs}>
        <CommitHeader
          onClick={e => this.handleRefresh(e)}
          lastUpdate={lastUpdate}
          count={count}
          repo={repo}
        />
        {
          isFetching ? <Spinner /> : commits && commits.map(commit => (
            <Commit key={commit.id} payload={commit} />
          ))
        }
        <Pagination
          onClickNext={() => this.onClickNext()}
          onClickPrev={() => this.onClickPrev()}
          next={nextPage == null}
          prev={prevPage == null}
        />
      </Container>
    );
  }
}


CommitViewer.propTypes = {
  dispatch: PropTypes.func.isRequired,
  history: PropTypes.object,  // eslint-disable-line react/forbid-prop-types
  commits: PropTypes.arrayOf(commitPropTypes),
  lastUpdate: PropTypes.string,
  isFetching: PropTypes.bool,
  nextPage: PropTypes.string,
  prevPage: PropTypes.string,
  count: PropTypes.number,
  repo: PropTypes.string,
  breadcrumbs: PropTypes.arrayOf(breadcrumbPropTypes),
};


CommitViewer.defaultProps = {
  commits: [],
  history: null,
  breadcrumbs: [],
  lastUpdate: null,
  isFetching: false,
  nextPage: null,
  prevPage: null,
  count: 0,
  repo: '',
};


const mapStateToProps = (state) => {
  const { selectedCommits } = state;
  const {
    isFetching,
    didInvalidate,
    errorMessage,
    commits,
    lastUpdate,
    nextPage,
    prevPage,
    count,
  } = selectedCommits || {
    isFetching: false,
    didInvalidate: false,
    errorMessage: '',
    commits: [],
    lastUpdate: null,
    nextPage: null,
    prevPage: null,
    count: 0,
  };

  return {
    isFetching,
    didInvalidate,
    errorMessage,
    commits,
    lastUpdate,
    nextPage,
    prevPage,
    count,
  };
};


export default connect(mapStateToProps)(CommitViewer);
