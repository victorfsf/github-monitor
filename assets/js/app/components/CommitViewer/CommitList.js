import React from 'react';
import PropTypes from 'prop-types';
import { commitPropTypes } from 'types';
import { NotFound, Spinner } from 'app/components';
import { Link } from 'react-router-dom';
import Commit from './Commit';
import './styles.scss';


const CommitList = (props) => {
  const { items, isFetching } = props;

  if (isFetching) {
    return <Spinner />;
  }

  if (!items || items.length === 0) {
    return (
      <NotFound title="Nothing to see here!" faClassName="fa fa-4x fa-frown-o">
        Come back after <Link to="/">adding some repositories</Link>.
      </NotFound>
    );
  }

  return (
    <div>
      {items.map(
        commit => (
          <Commit key={commit.id} data={commit} />
        ),
      )}
    </div>
  );
};


CommitList.propTypes = {
  items: PropTypes.arrayOf(commitPropTypes),
  isFetching: PropTypes.bool,
};


CommitList.defaultProps = {
  isFetching: false,
  items: [],
};


export default CommitList;
