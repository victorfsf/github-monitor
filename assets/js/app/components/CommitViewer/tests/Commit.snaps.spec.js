import React from 'react';
import renderer from 'react-test-renderer';
import Commit from 'app/components/CommitViewer/Commit';
import * as AvatarModule from 'app/components/CommitViewer/Avatar';
import * as ReactRouter from 'react-router-dom';
import * as utils from 'utils';

utils.formatDate = jest.fn(() => 'MMM D, YYYY [at] h:mm:ss A');

ReactRouter.Link = jest.fn(props => (
  <a href={props.to}>{props.children}</a>
));

AvatarModule.default = jest.fn(props => (
  <div id={props.id} name={props.name} />
));


describe('Commit', () => {
  let Component;
  let tree;

  test('renders with payload', () => {
    Component = renderer.create(
      <Commit
        data={{
          message: 'test',
          date: 'today',
          branch: 'master',
          repository: 'test/test',
          url: 'https://github.com/',
          sha: '132465789',
          author: {
            login: 'author',
            github_id: 1,
            name: 'Author Name',
          },
        }}
      />,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('does not render without payload', () => {
    global.console = {
      error: jest.fn(),
    };
    renderer.create(<Commit />);
    expect(global.console.error).toHaveBeenCalledTimes(2);
  });
});
