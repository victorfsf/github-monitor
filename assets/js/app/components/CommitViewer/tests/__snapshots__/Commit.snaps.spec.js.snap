// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commit renders with payload 1`] = `
<div
  className="my-3 p-3 bg-white rounded box-border"
>
  <div
    className="media"
  >
    <a
      href="https://github.com/author"
      target="blank_"
    >
      <div
        id={1}
        name="Author Name"
      />
    </a>
    <div
      className="media-body"
    >
      <div
        className="row"
      >
        <div
          className="col-md-8 col-sm-12"
        >
          <span
            className="title"
            title="test"
          >
            test
          </span>
        </div>
        <div
          className="col-md-4 col-sm-12"
        >
          <small
            className="float-md-right text-muted"
          >
            Apr 2, 2018 at 7:53:19 PM
          </small>
        </div>
      </div>
      <small
        className="row"
      >
        <div
          className="col-md-8 col-sm-12 pt-xs-2 pt-sm-2 pt-md-0"
        >
          <a
            href="https://github.com/author"
            target="blank_"
          >
            <i
              className="fa fa-user-o pr-1"
            />
            Author Name
          </a>
          <span
            className="pl-1 text-muted"
          >
            [author]
          </span>
          <i
            className="fa fa-code-fork pl-2 pr-1"
          />
          master
        </div>
        <div
          className="col-md-4 col-sm-12 pt-sm-2 pt-md-0"
        >
          <span
            className="float-md-right"
          >
            <a
              href="/commits/test/test"
            >
              test/test
            </a>
          </span>
        </div>
      </small>
      <small
        className="row"
      >
        <div
          className="col-12 pt-sm-2 pt-md-0"
        >
          <i
            className="fa fa-code pr-1"
          />
          <a
            href="https://github.com/"
            target="blank_"
          >
            132465789
          </a>
        </div>
      </small>
    </div>
  </div>
</div>
`;
